### DISCUSSION TITLE
Nonlinear solve did not converge due to DIVERGED_FNORM_NAN.

### DISCUSSION THREAD

**Discussion by RbmIjpb** [idaholab/moose#28192] [Q&A Modules: Solid mechanics]
Hello, 
I'm hoping someone can help me with my problem. 

I have a recurrent error from this code coming from 
`195       double _d_phi_d_J1 = theta * theta / (2 * phi * std::pow(_Y1[_qp] * _Y1[_qp] - theta, 2)) ;`
the intermediate messages i added to the code show the following : 

```
Before solveForTheta - J1: 0, J2: 0, _Y1[_qp]: 3e+06, _Y2[_qp]: 2e+06, _h[_qp]: 300
After solveForTheta - theta: 9e+12
Warning: theta exceeded maximum allowed value. Capping at 3.96e+12
f_dev calculation: 0
f_dev calculation: 0
Before solveForTheta - J1: 0, J2: 0, _Y1[_qp]: 3e+06, _Y2[_qp]: 2e+06, _h[_qp]: 300
After solveForTheta - theta: 9e+12
Warning: theta exceeded maximum allowed value. Capping at 3.96e+12
f_dev calculation: 0
f_dev calculation: 0
```

and i'm still getting

` Nonlinear solve did not converge due to DIVERGED_FNORM_NAN iterations 0`

the dbg message is this : 

`Thread 1 "lollipop-dbg" received signal SIGFPE, Arithmetic exception.
0x00007ffff38f9fb0 in NoGrowthCW::computeQpPK1Stress (this=0x555555c4a280)
    at /home/antilope/projects/lollipop/src/kernels/NoGrowthCW.C:195
195       double _d_phi_d_J1 = theta * theta / (2 * phi * std::pow(_Y1[_qp] * _Y1[_qp] - theta, 2)) ;`

It seems also that all the vectors created with multiply_vector_tensor have memory allocation problem, even after initializing `std::vector<double> a_bar(3)` : 

```
a_bar = {<std::_Vector_base<double, std::allocator<double> >> = {
    _M_impl = {<std::allocator<double>> = {<__gnu_cxx::new_allocator<double>> = {<No data fields>}, <No data fields>}, <std::_Vector_base<double, std::allocator<double> >::_Vector_impl_data> = {_M_start = 0x555555d98f40,
        _M_finish = 0x555555d98f58, _M_end_of_storage = 0x555555d98f58}, <No data fields>}}, <No data fields>}
```
Any help will be more than welcome and greately appreciated !! 

I added the header and source file here, in .txt.

[NoGrowthCWh.txt](https://github.com/user-attachments/files/16338456/NoGrowthCWh.txt)
[NoGrowthCW .txt](https://github.com/user-attachments/files/16338460/NoGrowthCW.txt)

**Comment by GiudGiud** [comment]
Hello

You could follow these instructions
https://mooseframework.inl.gov/application_development/debugging.html
and in the debugger print to screen each variable in
` double _d_phi_d_J1 = theta * theta / (2 * phi * std::pow(_Y1[_qp] * _Y1[_qp] - theta, 2)) ;`
until you find the one causing the exception

The debugger can use a breakpoint on that exception

Guillaume
